### PR TITLE
ZEPPELIN-312: fix a bug with blocking websocket broadcast

### PR DESCRIPTION
**Pull request by bzz**
### What is this PR for?
Replacing synchronization through critical section over the collection of sockets with the lock-free collection implementation  `java.util.concurrent.ConcurrentLinkedQueue`.
Synchronization was used to avoid parallel collection modifications, as the calls `.sendMessage()` in Jetty implementation of Websockets are thread-safe and can proceed concurrently.

### What type of PR is it?
Bug Fix

### Is there a relevant Jira issue?
https://issues.apache.org/jira/browse/ZEPPELIN-312

### How should this be tested?
See JIRA

### Questions:
* Does the licenses files need update? No
* Is there breaking changes for older versions? No
* Does this needs documentation? No

### CODE REVIEW

**Comment by bzz** [comment]
This is another attempt, after #319 
\cc @Leemoonsoo @zenglinxi0615 for review.

**Comment by Leemoonsoo** [comment]
CI failing looks irrelevant. LGTM

File: zeppelin-server/src/main/java/org/apache/zeppelin/server/ZeppelinServer.java [hunk 1]
```diff
@@ -63,37 +63,37 @@
 public class ZeppelinServer extends Application {
   private static final Logger LOG = LoggerFactory.getLogger(ZeppelinServer.class);
 
-  private SchedulerFactory schedulerFactory;
-  public static Notebook notebook;
-  public static NotebookServer notebookServer;
-  public static Server jettyServer;
+  public static Notebook NOTEBOOK;
+  public static NotebookServer NOTEBOOK_SERVER;
```
Comment (anthonycorbacho) [inline zeppelin-server/src/main/java/org/apache/zeppelin/server/ZeppelinServer.java]: I thinks this should be in camel
according to https://google.github.io/styleguide/javaguide.html#s5.2.5-non-constant-field-names

Comment (bzz) [inline zeppelin-server/src/main/java/org/apache/zeppelin/server/ZeppelinServer.java]: Thanks for point that out!
Or they have to be constant...

**Comment by bzz** [comment]
CI failure logged under https://issues.apache.org/jira/browse/ZEPPELIN-525

**Comment by bzz** [comment]
CI passes now, @anthonycorbacho review was addressed in bbbf8ae

**Comment by Leemoonsoo** [comment]
+1 Let's merge.

**Comment by zenglinxi0615** [comment]
@bzz ,sorry for my late response, and thanks for your work, it looks good for me!

**Comment by bzz** [comment]
Cool, thank you for your kind feedback guys! 
Merging if there is no further discussion.

**Comment by bzz** [comment]
@zenglinxi0615 could you please close #319 if the issue is solved?

**Comment by zenglinxi0615** [comment]
closed just now.

**Pull request closed without merging**